packages/template-retail-react-app/app/hooks/use-url-template.js

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useContext} from 'react'
+import {UrlTemplateContext} from '../contexts'
+
+/**
+ * Custom React hook to get the current site
+ * @returns {site: Object, setSite: function}
+ */
+const useUrlTemplate = () => useContext(UrlTemplateContext)
```

Comment:
To call `fillUrlTemplate()`, users would need to pass in a few parameters. How about we make it easier for them? So they need to pass in only 1 parameter, which is the url path. For example: `fillUrlTemplate('/login')` and that's it.

I think we can make this `useUrlTemplate` hook a bit smarter, such that it can automatically find out the `site` and `locale` values for you. I'm thinking something like this:

```suggestion
const useUrlTemplate = () => {
    const {fillUrlTemplate: originalFn} = useContext(UrlTemplateContext)
    const {site} = useSite
    const {locale} = useLocale

    const fillUrlTemplate = useCallback(
        (path, config) =>
            originalFn(path, site.alias || site.id, locale.alias || locale.id, config),
        [originalFn, site, locale]
    )
    return {fillUrlTemplate}
}
```

Reply:
The example code will not work. If we try to use `useSite` or `useLocale` inside `useUrlTemplate` we get the error `useSite` must be used within `SiteProvider`.

 The usage of `useCallback` hook doesn't seem to be necessary to override some of the arguments.
 
But I like the idea of passing fewer parameters to `fillUrlTemplate`.

I end up using the `useContext` hook and "merging" Site, Locale and UrlTemplate Providers into a `MultiSiteProvider`.